lib/chatkit/client.rb

```diff
@@ -769,11 +804,54 @@ def update_permissions_for_role(options)
       })
     end
 
+    def upload_attachment(token, room_id, part)
+      body = part[:file]
+      content_length = body.length
+
+      if content_length <= 0
+        raise Chatkit::MissingParameterError.new("File contents size must be greater than 0")
+      end
+
+      attachment_req = {
+        content_type: part[:type],
+        content_length: content_length
+      }
+
+      attachment_response = api_request({
+        method: "POST",
+        path: "/rooms/#{CGI::escape room_id}/attachments",
+        body: attachment_req,
+        jwt: token
+      })
+
+      url = attachment_response[:body][:upload_url]
+      connection = Excon.new(url, :omit_default_port => true)
+      upload_response = connection.put(:body => body)
+
+      if upload_response.status != 200
+        error = {message: "Failed to upload attachment",
+                 response_object: upload_response
+                }
+        raise Chatkit::UploadError.new(error)
+      end
+
+      attachment_response[:body][:attachment_id]
+    end
+
+    def get_optional_params(field_names, options)
+      params = {}
+
+      field_names.each { |name|
+        params[name] = options[name] unless options[name].nil?
+      }
+      params
```

Comment:
Suggested:

```
params.reject(&:nil?)
```

Which is sugar for:

```
params.reject { |p| p.nil? }
```

Ruby has good map / filter / fold (but with potentially unexpected names because of smalltalk heritage) and they are strongly encouraged. 

They also have loads of aliases and negations, the above is sugar for

```
params.select { |p| !p.nil? }
```

See the monster list here: https://ruby-doc.org/core-2.4.0/Enumerable.html

Reply:
Um, I misread that one.